expression/builtin_time.go

```diff
@@ -5559,6 +5559,9 @@ func (b *builtinAddStringAndStringSig) evalString(row chunk.Row) (result string,
 	if isNull || err != nil {
 		return "", isNull, err
 	}
+	if !isDuration(arg1Str) {
```

Comment:
> Test case fails
> 
> ```
> [2021-08-12T09:34:14.503Z] FAIL: builtin_time_test.go:903: testEvaluatorSuite.TestAddTimeSig
> [2021-08-12T09:34:14.503Z] 
> [2021-08-12T09:34:14.503Z] builtin_time_test.go:1012:
> [2021-08-12T09:34:14.503Z]     c.Assert(len(warnings), Equals, i+1+beforeWarnCnt)
> [2021-08-12T09:34:14.503Z] ... obtained int = 8
> [2021-08-12T09:34:14.503Z] ... expected int = 9
> [2021-08-12T09:34:14.503Z] 
> ```

Here is the reason, if `arg1Str` is not a MySQL `Time`, it just return without error.